File: manual/data/python/src/function_joblib_delayed.py
from joblib import delayed, Parallel


class DelayedExample:
    def apply_delayed_function(self, x: int, y: int) -> str:
        add_func = delayed(lambda a, b: a + b)

        result = Parallel(n_jobs=1)([add_func(x, y)])

        return str(result[0])

File: manual/data/java/src/function_joblib_delayed.java
package com.example;

import com.google.common.util.concurrent.ListenableFuture;
import com.google.common.util.concurrent.ListeningExecutorService;
import com.google.common.util.concurrent.MoreExecutors;
import java.util.concurrent.Callable;
import java.util.concurrent.Executors;
import java.util.concurrent.Future;

public class DelayedExample {

    public String apply_delayed_function(int x, int y) {
        ListeningExecutorService executorService = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(1));

        try {
            Callable<Integer> addFunc = () -> x + y;
            ListenableFuture<Integer> future = executorService.submit(addFunc);
            return String.valueOf(future.get());
        } catch (Exception e) {
            return null;
        } finally {
            executorService.shutdown();
        }
    }
}